src/note.c

```diff
@@ -162,8 +169,11 @@ void scrotNoteNew(char const *const format)
             while (c) {
                 token = c;
 
-                const int color = optionsParseRequireRange(
-                        optionsParseRequiredNumber(c), 0 ,255);
+                const int color = options_parsenum(c, 0, 255, &errmsg);
```

Comment:
The last color digit is not being parsed, when using the following: `...-c 0,255,0,150 -t 'Hi'"` you get an error:
```bash
scrot: option --note: color '150 -t 'Hi'' is not a number
```
- The problem is in function `options_parsenum`, when it detects if it is a number or not. The solution would be what the manpage of `strtoll` indicates.

```diff
-    } else if (*end != '\0') {                                                                  
+    } else if (str == end) {         
```

- The other solution is to replace the space character just before calling the function:
```diff
+                char *const space = strchr(c, ' ');
+
+                if (space) *space = '\0';
+
                 const int color = options_parsenum(c, 0, 255, &errmsg);
                 if (errmsg) {
                         errx(EXIT_FAILURE, "option --note: color '%s' is %s", c,
                             errmsg);
                 }
 
+                if (space) *space = ' ';
```

---
I prefer this second solution because with `if (*end != '\0') ` it does not allow for example a string like `22a`, I think it is better.

Reply:
That will have unwanted fallout, if you pass a color like "1 2,3,4" the error message won't look quite right. The right solution would probably use strspn() to skip over all the numbers and commas and check if a space follows, I'll have a look at it later.

Reply:
I don't understand what error you say. Your last commit 4c318bf also allows the form '1 2,3,4' without showing any errors.
And have 2 problems: first it doesn't allow the color to be used as the last parameter: `-t 'Hi' -c 255` and second with: `-c a,255`  say: `scrot: option --note: suboption -c: red channel '' is the null string`
but with my patch it works correctly :` scrot: option --note: color 'a' is not a number` and it does not change the behavior of the option `--note` that allows to place the color at the end.

Reply:
The truth is that I do not like https://github.com/resurrecting-open-source-projects/scrot/commit/4c318bfa63b461b220c8580b84cf4f107174996a.
It's not "fair" for a function to change so much because of behavior outside of it, which was working fine.

For the problem that resulted in changing `optionsParseRequireRange` to `options_parsenum` my 3 lines of code are enough for now.
If a lot needs to be corrected, a new PR will open soon after I merge this one.

**I would really like this PR to be in the next `release` that will be released in the next few days** :(

Reply:
I've squashed the commits and done the requested changes.